force-app/main/default/classes/PMT_RefundService.cls

```diff
@@ -226,12 +216,15 @@ public inherited sharing class PMT_RefundService {
         );
     }
 
-    private void buildRefundRecord(npe01__OppPayment__c originalPayment) {
+    public npe01__OppPayment__c buildRefundRecord(npe01__OppPayment__c originalPayment, Decimal refundAmount) {
```

Comment:
To make this more clear, could we make a second method for Full refunds that doesn't take a refundAmount? Passing `null` for Full refunds just doesn't feel right to me...

Reply:
Instead of adding a second method for no refund amount. I change the class that is calling the method to pass in a refund amount instead of null. This way the test methods will be more align to the actual use case